Updates for C9 (React!)

droberts-sea opened the pull request:
- Update verbiage to React instead of Backbone
- Add the Router to the requirements (this is something they will have to research)
- Somewhat expand the requirements, so they have to list customers and check out movies. This project was too easy last cohort and we've got an extra day this time, so I believe this is warranted.
- Add some optionals around DevOps stuff

~~The Rails API itself is OK, though the documentation in the README has drifted a little~~ (edit: this is now fixed).

Still need to go through and do an implementation in React, to tease out the snarls.

Commit:
Start working the README

Commit:
Expand front-end requirements

Commit:
Re-work the optionals to highlight deployment and CI

Commit:
Add in another optional

Commit:
Reorganize design considerations

Commit:
Section on back-end requirements

Commit:
Update PR template

Commit:
Update feedback document

Commit:
Remove space for optionals

Commit:
Add link to feedback.md from README

CheezItMan commented on .github/PULL_REQUEST_TEMPLATE:
```diff
@@ -4,10 +4,11 @@ Congratulations! You're submitting your assignment!
+Describe a DOM event your application handled |
+Did you use any functional stateless components? What for? |
+Did you use any container components? What for? |
+Do you have any recommendations on how we could improve this project for the next cohort? |
```
Add a link to the API PR

CheezItMan reviewed (approved):
Looks good, I like the fact that we're having them research together React Router.

Commit:
Add secton for link to API PR to PR_TEMPLATE

Commit:
Update the README from working through the solution

Commit:
Run create-react-app

We really wanted to make the students do this, but you can't run CRA in
a non-empty directory because Mark Zuckerberg is a big dummy. See
https://github.com/facebook/create-react-app/issues/334

Commit:
Clarification on how many movies you can check out
